### DISCUSSION TITLE
BUG: RayTest with Softbody returns no index(PyBullet)

### DISCUSSION THREAD

**Discussion by Simon-Steinmann** [bulletphysics/bullet3#3739] [General]
Hi, 
After 2days of researching, source code inspection and hours of github issues reading, I'm at my wit's end.
My goal is, to simulate a robot with a suctiono gripper, to pic up textiles. I got so far as to add and remove softBodyAnchors, however, the issue is, that I have to know the nodeID of where to attach the anchor. I tried getting that ID by using 
`p.rayTestBatch(pos1, pos2)`, but it only returns the object ID and `-1 `as the nodeID. 

I managed to get the mouse picking to work on the textile, making me wonder, whether something like this could be used. Applying forces to the softBody, where a rayTest intersects it. is this possible somehow?

I would really appreciate it, if someone could answer any of these questions, propose a different solution or point me in the right direction.

**Comment by erwincoumans** [comment]
I just spend a few days improving the deformables/soft body, and also allowed to get the mesh data of the simulation mesh.
![image](https://user-images.githubusercontent.com/725468/92980485-52607880-f44b-11ea-8b10-8561d8435dfd.png)
Will report when the example/pull request is there.

**Comment by Simon-Steinmann** [comment]
Thank you so much. This is highly appreciated. If I can do anything to help or test, let me know. My c++ skills are limited (can read it a bit, but not proficient coder). But I'm good with python. Will this allow one to get the nodeID of a softbody?
